[MNG-7360] Fix xml transformation to ensure proper context

gnodet opened the pull request:
As hinted by @rfscholte , the `FastForwardFilter` is supposed to handle those problems by disabling the filters when parsing  xml config nodes.  I've added a unit test and investigated in that direction, which leads to this new PR.  It looks much simpler than #697, so I think it's better to keep that one.

rfscholte commented on maven-model-transform/src/main/java/org/apache/maven/model/transform/pull/BufferingParser.java:
```diff
@@ -430,7 +430,7 @@ public int nextToken() throws XmlPullParserException, IOException
             }
             int currentEvent = xmlPullParser.nextToken();
-            if ( accept() )
+            if ( disabled || accept() )
```
This looks like to conflicting states, which makes me wonder if we can always return currentEvent.
If current expression is required, we need to think of better naming.

gnodet replied:
Maybe rename `disabled` to `bypass` ?
Or `accept()` to `canYieldCurrentEvent()` ?
Or both...

mthmulders replied:
I like `bypass` - and I'd like it even more if the variable name would be in the same terminology as the class Javadoc. So maybe we should rephrase the Javadoc a bit and say something like "This filter will bypass all following filters and write directly to the output."

mthmulders commented on maven-model-transform/src/test/java/org/apache/maven/model/transform/ParentXMLFilterTest.java:
```diff
@@ -33,30 +33,55 @@
+    }
+
+    @Test
+    public void testWithFastForward()
```
Just to check: would it make sense to also add a test for the opposite case, i.e. where the ParentXMLFilter is _not_ supposed to bypass other filters?

cstamas commented:
Restarted GH ITs as I broke and @slawekjaranowski fixed ITs 😄

mthmulders commented:
Although the changes look sane, I'm not sure we're on the right path. I've tried building this branch of Maven manually (succeeded) and then use that `mvn` to build Camel. Still fails:

```bash
mvn -V validate
Apache Maven 4.0.0-alpha-1-SNAPSHOT (c02b5f38d1dbc39db5b5360bed5c5807c3c6fa2a)
Maven home: /tmp/maven-gnodet/apache-maven-4.0.0-alpha-1-SNAPSHOT
Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.3.1", arch: "x86_64", family: "mac"
[INFO] BuildTimeEventSpy is registered.
[INFO] Scanning for projects...
[ERROR] Some problems were encountered while processing the POMs
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project (/Users/maarten/Code/open-source/camel/pom.xml) has 1 error
[ERROR]     Unable to transform pom: Not a regular file: /Users/maarten/Code/open-source/pom.xml
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the '-e' switch
[ERROR] Re-run Maven using the '-X' switch to enable verbose output
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

gnodet commented:
Thx for the check ! I'll work on an IT and a proper fix on monday.

gnodet commented:
IT added with https://github.com/apache/maven-integration-testing/pull/167

gnodet commented:
> Although the changes look sane, I'm not sure we're on the right path. I've tried building this branch of Maven manually (succeeded) and then use that `mvn` to build Camel. Still fails:
> 
> ```shell
> mvn -V validate
> Apache Maven 4.0.0-alpha-1-SNAPSHOT (c02b5f38d1dbc39db5b5360bed5c5807c3c6fa2a)
> Maven home: /tmp/maven-gnodet/apache-maven-4.0.0-alpha-1-SNAPSHOT
> Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
> Default locale: en_US, platform encoding: UTF-8
> OS name: "mac os x", version: "12.3.1", arch: "x86_64", family: "mac"
> [INFO] BuildTimeEventSpy is registered.
> [INFO] Scanning for projects...
> [ERROR] Some problems were encountered while processing the POMs
> [ERROR] The build could not read 1 project -> [Help 1]
> [ERROR]
> [ERROR]   The project (/Users/maarten/Code/open-source/camel/pom.xml) has 1 error
> [ERROR]     Unable to transform pom: Not a regular file: /Users/maarten/Code/open-source/pom.xml
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the '-e' switch
> [ERROR] Re-run Maven using the '-X' switch to enable verbose output
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
> ```
> 
> I've attached a debugger, but it seems the `FastForwardFilter` is never hit.

I've added an IT and fixed the behavior.

gnodet commented:
@mthmulders would you mind approving the PR for the record ?

mthmulders commented:
> @mthmulders would you mind approving the PR for the record ?

My pleasure!
